src/app/qgisapp.cpp

```diff
@@ -12344,6 +12335,19 @@ QMenu *QgisApp::createPopupMenu()
   return menu;
 }
 
+void QgisApp::notifyUser( QString title, QString message )
```

Comment:
Hi. I agree with @m-kuhn's suggestions here. `showSystemNotification` is more semantically descriptive, and anything that Qt seems to have 'forgot' in their libraries, or is OS-specific code and can be wrapped in an abstract interface, should really go in a native wrapper setup. Even though Qt is missing a simple API for sending OS notifications to the user, but still provides a hack-ish way of doing so via `QSystemTrayIcon`, there are possibly better native APIs for sending OS notifications, e.g. like [Obj-C++ calls for macOS](https://wiki.qt.io/How_to_use_OS_X_Notification_Center).

Once the abstracted call is in the native wrapper setup, the OS-specific implementation can be upgraded later, if needed.

However, the wrapper classes are not merged yet, so I think merging @elpaso's change here is a reasonable first step.